xbmc/music/infoscanner/MusicInfoScanner.cpp

```diff
@@ -919,6 +923,58 @@ int CMusicInfoScanner::RetrieveMusicInfo(const std::string& strDirectory, CFileI
   return numAdded;
 }
 
+void CMusicInfoScanner::CheckBoxSets(VECALBUMS &albums)
+{
+  /* Strategy: Having scanned the tags of the albums to add to the library,
+   * check for disc subtitles. If an album only has one subtitled disc, we don't
+   * consider it a boxset as its most likely a 'live'version of the album or a 'bonus disc'.
+   * Single disc subtitles are still added to the database so they can be displayed by
+   * skins at appropriate times,  If 'boxset' was added to the albumreleasetype then add the
+   * album regardless of any other rules. */
+
+  std::string m_oldDiscSubtitle;
+  int count;
+  int discno;
+  int old_discno = 0;
+  for(auto& album : albums)
+  {
+    if (album.bCompilation  && !album.bBoxedSet)
+    {
+      album.bBoxedSet = false;
```

Comment:
This only reached when !album.bBoxedSet anyway, setting does nothing?

Reply:
Yeah, setting the bool to false again is pointless, it should just be continue. I'll take line 943 out.  The check is important though, because without it, Kodi will start to create a boxset for a compilation (and it will show in the first node) but there won't be any discs associated with it.